extensions/git/src/git.ts

```diff
@@ -269,7 +269,7 @@ export class GitError {
 			this.error = void 0;
 		}
 
-		this.message = this.message || data.message || 'Git error';
+		this.message = this!.message || data.message || 'Git error';
```

Comment:
Why `this!`? And why not on the left?

Reply:
`!.` is generally the not-null type assertion. This case is weird though and I'm sort of abusing the `!.`

Root cause is that TS is smart and will flag code like:

```js
class Foo {
   constructor() {
       this.message = this.message.toString()
   }
}
```

as a potential error. In the git extension example however, it doesn't matter if we have assigned to `this.message` before evaluating the logical or expression. The `!.` tells TS to not care about this potential error since it doesn't actually matter.

An alternative fix would be to explicitly set `this.message` to `undefined` or `''` before this line